generators/server/index.js

```diff
@@ -956,7 +979,8 @@ module.exports = JhipsterServerGenerator.extend({
             this.template(SERVER_MAIN_SRC_DIR + 'package/security/_UserDetailsService.java', javaDir + 'security/UserDetailsService.java', this, {});
             this.template(SERVER_MAIN_SRC_DIR + 'package/security/_UserNotActivatedException.java', javaDir + 'security/UserNotActivatedException.java', this, {});
 
-            if (this.authenticationType === 'jwt') {
+
+            if (this.authenticationType == 'jwt' && this.applicationType !== 'uaa') {
```

Comment:
this should be `this.applicationType !== 'uaa' && this.authenticationType == 'jwt'` for readability